Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Processing/ImageObjectImpl.cpp

```diff
@@ -352,6 +353,63 @@ namespace ImageProcessingAtom
         return (float)avg;
     }
 
+    AZ::Color CImageObject::ComputeLinearAlphaWeightedAverageColor() const
+    {
+        // Check if we are in a compressed or non-linear (sRGB) space
+        if (!CPixelFormats::GetInstance().IsPixelFormatUncompressed(m_pixelFormat)
+            || HasImageFlags(EIF_SRGBRead))
+        {
+            IImageObjectPtr clone(Clone(1)); // Only most detailed mip is needed
+            IImageObjectPtr linear = GetUncompressedLinearImage(clone);
+            return linear->ComputeLinearAlphaWeightedAverageColor();
+        }
+
+        // We are already in a linear, uncompressed space
+        AZ_Assert(CPixelFormats::GetInstance().IsPixelFormatUncompressed(m_pixelFormat),
+                "Expected an uncompressed image")
+        AZ_Assert(!HasImageFlags(EIF_SRGBRead), "Expected a linearly-encoded image")
+
+        // Accumulate weighted pixel colors and alpha of the top mip
+        double weightedSum[3] = { 0.0, 0.0, 0.0 };
+        double alphaSum = 0;
+
+        // Create pixel operation function to access pixel data
+        IPixelOperationPtr pixelOp = CreatePixelOperation(m_pixelFormat);
+
+        // Get count of bytes per pixel for images
+        AZ::u32 pixelBytes = CPixelFormats::GetInstance().GetPixelFormatInfo(m_pixelFormat)->bitsPerBlock / 8;
+
+        // Only calculate mip 0
+        AZ::u32 mip = 0;
+        AZ::u8* pixelBuf;
+        AZ::u32 pitch;
+        GetImagePointer(mip, pixelBuf, pitch);
+        const AZ::u32 pixelCount = GetPixelCount(mip);
+
+        float color[3];
+        float alpha;
+        for (AZ::u32 i = 0; i < pixelCount; ++i, pixelBuf += pixelBytes)
+        {
+            pixelOp->GetRGBA(pixelBuf, color[0], color[1], color[2], alpha);
+            // Alpha-weighted sum for the r,g,b channels
+            weightedSum[0] += alpha * color[0];
+            weightedSum[1] += alpha * color[1];
+            weightedSum[2] += alpha * color[2];
+            alphaSum += alpha;
+        }
+
+        if (alphaSum == 0)
+        {
+            // Entire alpha channel was zero -> return zero to avoid div by 0
+            return Color(0.0f);
+        }
+
+        return Color((float) weightedSum[0] / alphaSum,
```

Comment:
I think we avoid c-style cast (but then the code above does that ...) could you consider `aznumeric_cast`?